pyat/at/plot/generic.py

```diff
@@ -0,0 +1,138 @@
+        baseplot divides the region of interest of ring into small elements,
+        calls the specified function to get the plot data and calls matplotlib
+        functions to generate the plot.
+        By default it creates a new figure for the plot, but if provided with
+        axes objects it can be used as part of a GUI
+
+        PARAMETERS
+            ring            Lattice object
+            plot_function   specific data generating function to be called
+
+            All other positional parameters are sent to the plotting function
+
+            plot_function is called as:
+
+            title, left, right = plot_function(ring, refpts, *args, **kwargs)
+
+            and should return 2 or 3 output:
+
+            title   plot title or None
+            left    tuple returning the data for the main (left) axis
+               left[0]   y-axis label
+               left[1]   xdata: (N,) array (s coordinate)
+               left[2]   ydata: iterable of (N,) or (N,M) arrays. Lines from a
+                         (N, M) array share the same style and label
+               left[3]   labels: (optional) iterable of strings as long as ydata
+            right   tuple returning the data for the secondary (right) axis
+                    (optional)
+
+        KEYWORDS
+            s_range=None    plot range, defaults to the full ring
+            axes=None       axes for plotting as (primary_axes, secondary_axes)
+                            Default: create new axes
+            slices=400      Number of slices
+            legend=True     Show a legend on the plot
+            dipole={}       Dictionary of properties overloading the default
+                            properties of dipole representation.
+                            See 'plot_synopt' for details
+            quadrupoles={}  Same definition as for dipoles
+            sextupoles={}   Same definition as for dipoles
+            multipoles={}   Same definition as for dipoles
+            monitor={}      Same definition as for dipoles
+
+            All other keywords are sent to the plotting function
+
+        RETURN
+            left_axes       Main (left) axes
+            right_axes      Secondary (right) axes or None
+            synopt_axes     Synoptic axes
+        """
+
+        def plot1(ax, yaxis_label, x, y, labels=()):
+            lines = []
+            for y1, prop, label in zip(y, props, chain(labels, repeat(None))):
+                ll = ax.plot(x, y1, **prop)
+                if label is not None:
+                    ll[0].set_label(label)
+                lines += ll
+            ax.set_ylabel(yaxis_label)
+            return lines
+
+        def labeled(line):
+            return not line.properties()['label'].startswith('_')
+
+        # extract baseplot arguments
+        s_range = kwargs.pop('s_range', None)
+        slices = kwargs.pop('slices', SLICES)
+        axes = kwargs.pop('axes', None)
+        legend = kwargs.pop('legend', True)
```

Comment:
Done. Defaulting to `True` also blocks in interactive mode, which I am using, so I set the default to `False`, since I do not see the interest of blocking.